src/main/java/book/store/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+import book.store.onlinebookstore.repository.user.UserRepository;
+import book.store.onlinebookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Your shopping cart is empty"));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+
+        Order newOrder = initNewOrder(requestDto, userId);
+        orderRepository.save(newOrder);
+
+        BigDecimal total = BigDecimal.ZERO;
+        Set<OrderItem> orderItems = cartItems.stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = orderItemMapper.fromCartItemToOrderItem(cartItem);
+                    orderItem.setPrice(
+                            orderItem.getPrice()
+                            .multiply(new BigDecimal(orderItem.getQuantity()))
+                    );
+                    newOrder.setTotal(total.add(orderItem.getPrice()).add(newOrder.getTotal()));
+                    orderItem.setOrder(newOrder);
+                    return orderItemRepository.save(orderItem);
+                }).collect(Collectors.toSet());
+        newOrder.setOrderItems(orderItems);
+
+        shoppingCartRepository.deleteById(userId);
+        return orderMapper.toDto(newOrder);
+    }
+
+    @Override
+    public List<OrderDto> getOrders(Long userId, Pageable pageable) {
+        List<Order> allByUserId = orderRepository.findAllByUserId(userId, pageable);
+        return allByUserId.stream().map(orderMapper::toDto).toList();
+    }
+
+    @Override
+    @Transactional
+    public List<OrderItemDto> getOrderItemsByOrderId(Long userId, Long orderId) {
+        Order order = orderRepository.findByIdAndUserId(orderId, userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order by id "
+                        + orderId));
+        return order.getOrderItems().stream().map(orderItemMapper::toDto).toList();
+    }
+
+    @Override
+    public OrderItemDto getItemById(Long userId, Long orderId, Long itemId) {

```

Comment:
3 GET methods - only 1 with `@Transactional` 

why ? :)

if you wish to use transactions in read method as well - consider annotating your service with 

`@Transactional(readOnly = true)`:
```
@Transactional(readOnly = true)
Service {

getMethod() {...}

@Transactional // to override read-only
updateMethod(){...}
}
```